### PR TITLE
reverse_words new challenge

### PR DESCRIPTION
**Pull request by firemanxbr**
New Challenger added.

### CODE REVIEW

File: tests/test_reverse_words.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import unittest
+
+from warzone.utils.reverse_words import reverse_words
+
+class Test(unittest.TestCase):
+    result = reverse_words(value='I am a software engineer. But,, I am also an operations engi.neer.')
+
+    def test_result(self):
+        self.assertEqual('I ma a erawtfos reenigne. tuB,, I ma osla na snoitarepo igne.reen.', self.result)
```
Comment (apahim) [inline tests/test_reverse_words.py]: The acquired goes first, the expected goes after.

File: warzone/utils/reverse_words.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    """
+    Creates a mirror(reverse) the words respecting the punctuation.
+
+    input = 'I am a software engineer. But,, I am also an operations engi.neer.'
+    expected = 'I ma a erawtfos reenigne. tuB,, I ma osla na snoitarepo igne.reen.'
+    """
+    match = re.finditer('[a-zA-Z]+', value)
+
+    for item in match:
+        value = value[:item.start()] + item.group()[::-1] + value[item.end():]
```
Comment (apahim) [inline warzone/utils/reverse_words.py]: Maybe using f-strings will make the syntax more 'modern' ? :)

**Review by apahim — CHANGES_REQUESTED** [review]

File: warzone/utils/reverse_words.py (hunk 2 above)
Comment (firemanxbr) [reply warzone/utils/reverse_words.py]: I did try modernize it with f-strings but I don't found a solution that allow us to load the variabeles and susing slicing togheter. Please can you show an example of it?

Comment (apahim) [reply warzone/utils/reverse_words.py]: `value = f'{value[:item.start()]}{item.group()[::-1]}{value[item.end():]}'`

**Comment by apahim** [comment]
Also, there's an unstable test making the CI fail from time to time. You can just remove that test.

**Comment by firemanxbr** [comment]
@apahim currently we have two testings failing:

$ python3 --version
Python 3.7.3rc1

marcelo@MININT-8OQK3F7:/mnt/c/Users/mabarbo/Documents/repositories/warzone$ python3 tests/run.py 
test_format (test_generate_hosts.Test) ... ok
test_size (test_generate_hosts.Test) ... ok
test_args (test_hashtable.TestHashTable) ... ok
**test_get_item_time (test_hashtable.TestHashTable) ... FAIL**
test_items (test_hashtable.TestHashTable) ... ok
test_keys (test_hashtable.TestHashTable) ... ok
test_repr (test_hashtable.TestHashTable) ... ok
test_table_resize (test_hashtable.TestHashTable) ... ok
test_args (test_linkedlist.TestLinkedList) ... ok
test_clear (test_linkedlist.TestLinkedList) ... ok
test_delete (test_linkedlist.TestLinkedList) ... ok
test_extend (test_linkedlist.TestLinkedList) ... ok
test_list_append (test_linkedlist.TestLinkedList) ... ok
test_list_length (test_linkedlist.TestLinkedList) ... ok
test_node_eq_ne (test_linkedlist.TestLinkedList) ... ok
test_qe_ne (test_linkedlist.TestLinkedList) ... ok
test_remove (test_linkedlist.TestLinkedList) ... ok
test_repr (test_linkedlist.TestLinkedList) ... ok
test_sub_list (test_linkedlist.TestLinkedList) ... ok
**test_reverse_words (unittest.loader._FailedTest) ... ERROR**

======================================================================
ERROR: test_reverse_words (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: test_reverse_words
Traceback (most recent call last):
  File "/usr/lib/python3.7/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.7/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/mnt/c/Users/mabarbo/Documents/repositories/warzone/tests/test_reverse_words.py", line 3, in <module>
    from warzone.utils.reverse_words import reverse_words
ModuleNotFoundError: No module named 'warzone.utils.reverse_words'


======================================================================
FAIL: test_get_item_time (test_hashtable.TestHashTable)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/c/Users/mabarbo/Documents/repositories/warzone/tests/test_hashtable.py", line 61, in test_get_item_time
    self.assertAlmostEqual(time_taken_low_index, time_taken_high_index, places=1)
AssertionError: 0.2505016000000069 != 0.4799940999996579 within 1 places (0.22949249999965105 difference)

----------------------------------------------------------------------
Ran 20 tests in 1.312s

FAILED (failures=1, errors=1)

**Comment by apahim** [comment]
True. The one I'm talking about is the `test_get_item_time`. You can just remove it. You still have to fix the test you're introducing here though.

**Comment by apahim** [comment]
@firemanxbr the `test_reverse_words` test is not failing on the CI. Maybe you just have to reinstall the package in your local machine? Are installing an editable version (`pip install -e .` or `python setup.py develop`)?

**Review by apahim — APPROVED** [review]

**Pull request merged**